### DISCUSSION TITLE
Two types of grains on histogram

### DISCUSSION THREAD

**Discussion by NikitaSakharov** [mtex-toolbox/mtex#978] [Ask Anything]
Hi, everyone!

Could anyone help me, please? 

What I want to do is to plot two or more grains size distributions on one histogram.
How can I do this? Unfortunately I have not found such an example in documentation.

Thanks in advance!

And by the way, is there any way I can change histogram appearance (legend, title, axes names, colors)? Is it possible to make MtexFigure which contains several histograms (just in the way I can do it with Mtexfigure which contains plots)?

**Comment by kilir** [comment]
Hi Nikita.,
Matlabs `bar()` plot can do that. You can extract the area weighted data from the handle (havea look at the help) to `grain2d/histogram()`. You'd then need to generate bin centers ,  bincounts will come from the handle and plot those with `bar()`.

However, `grain2d/histogram()` only take the number of bins but as of now not fixed bins as an input - which you'd need if you want to display the results together, if I understand you correctly. Either you do the weighting yourself and use `histcounts()`  or simply add the relevant option to `grain2d/histogram()` to allow putting fixed bins as an option.

Hope this helps,
Rüdiger

**Reply by NikitaSakharov** [reply]
Hello Rudiger.

I think adding option to "grain2d/histogram()" function will be too difficult for me. So i'll need to try some workaround using bar().

Thanks a lot!
